OpenRA.Mods.Common/Traits/Repairable.cs

```diff
@@ -142,15 +142,20 @@ public Actor FindRepairBuilding(Actor self)
 
 		static void TryCallTransport(Actor self, Target target, Activity nextActivity)
 		{
-			var transport = self.TraitOrDefault<ICallForTransport>();
-			if (transport == null)
+			var transports = self.TraitsImplementing<ICallForTransport>();
+			if (!transports.Any())
 				return;
 
 			var targetCell = self.World.Map.CellContaining(target.CenterPosition);
-			if ((self.CenterPosition - target.CenterPosition).LengthSquared < transport.MinimumDistance.LengthSquared)
-				return;
+			var delta = (self.CenterPosition - target.CenterPosition).LengthSquared;
+
+			foreach (var t in transports)
+			{
+				if (delta < t.MinimumDistance.LengthSquared)
+					continue;
 
-			transport.RequestTransport(self, targetCell, nextActivity);
+				t.RequestTransport(self, targetCell, nextActivity);
```

Comment:
Is this a possible race condition? I'm not super familiar with the transport code, but there was no possibility of two transports arriving on the same tick before.